content/docs/connect/choose-connection.md

```diff
@@ -0,0 +1,163 @@
+enableTableOfContents: true
+---
+
+When setting up your application’s connection to your Neon Postgres database, you need to make two main choices: the type of driver and the type of connection. This flowchart will guide you through these options.
+
+## Choosing your connection type: flowchart
+
+![choose your connection type](/docs/connect/choose_connection.png)
+
+## Choosing your connection type: drivers and pooling
+
+### Step 1: Choose your Postgres driver
+
+Your first choice is which driver to use. If working in a serverless environment, use the [Neon Serverless Driver](/docs/serverless/serverless-driver). It's built to handle dynamic workloads with high variability in traffic &#8212; for example, Vercel Edge Functions or Cloudflare Workers.
+
+If you don't need serverless, use a traditional TCP-based Postgres driver. For example, if you’re using Node.js with a framework like Next.js, you can add the `pg` client to your dependencies, which acts as the TCP-based Postgres driver.
+
+### Step 2: If serverless, choose HTTP or WebSocket connections
+
+Neon offers two types of serverless connections: HTTP and WebSocket.
+
+- **Use HTTP for**: One-shot queries and short-lived operations where connection setup and teardown costs are minimal.
+- **Use WebSocket for**: [Transactions](/docs/postgresql/query-reference#transactions) and operations requiring persistent connections over a session. WebSocket connections provide lower latency for multiple queries in the same session.
+
+<Admonition type="note">
+We are working on automatic switching between HTTP and WebSocket to as needed. Check our [roadmap](/docs/introduction/roadmap) to see what's coming soon and our Friday [Changelog](/docs/changelog) for the features-of-the-week.
+</Admonition>
+
+### Step 3: Next, choose your connection type: direct or pooled
+
+You then need to decide whether to use direct connections or pooled connections using PgBouncer:
+
+- **In general, use pooled connections whenever you can**: Pooled connections can efficiently manage high numbers of concurrent connections, supporting up to 10,000 connection. This 10,000 connection limit is most useful for serverless applications and application-side connection pools that have many open connections, but infrequent and/or short transactions.
+- **Use direct (unpooled) connections if you need persistent connections**: If your application is focused mainly on tasks like migrations or administrative operations that require stable and long-lived connections, use an unpooled connection.
+
+<Admonition type="note">
+The use of connection pooling is not a magic bullet. PgBouncer has a `default_pool_size` of 64, meaning it can handle only 64 direct connections to the database at any one time. When your application connects to PgBouncer, it joins a large pool, but only 64 connections are processed at a time. Long operations can cause these connections to time out or close before tasks are completed.
```

Comment:
```suggestion
Connection pooling is not a magic bullet. It can manage 10,000 concurrent connections from your application to PgBouncer, but PgBouncer's default_pool_size setting is set to 64 by default, which means connections from PgBouncer to Postgres are limited to 64 for each user-database pair. For example, Postgres user `alex` can hold up to 64 connections to a single database at one time.
```

Reply:
Could it be a combination of connections held open for too long, or is it also volume?